shared/Microsoft.AspNetCore.Routing.UrlMatchingTree.Sources/TreeEnumerator.cs

```diff
@@ -114,6 +118,194 @@ public void Reset()
             _stack.Clear();
             Current = null;
         }
-    }
 
+        internal static void AddEntryToTree(UrlMatchingTree tree, InboundRouteEntry entry)
```

Comment:
This should move to the `UrlMatchingTree` class. This isn't the right home for it.